### PR TITLE
Acceptance test for multiple public links with same name for same file/folder

### PR DESCRIPTION
**Pull request by jasson99**
## Description
Acceptance test for multiple public links with same name for same file/folder

## Related Issue

- Fixes issue #2526

## How Has This Been Tested?

-

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [ ] ...

### CODE REVIEW

File: tests/acceptance/pageObjects/FilesPageElement/publicLinksDialog.js [hunk 1]
```diff
@@ -381,6 +381,18 @@ module.exports = {
         message = result.value
       })
       return message
+    },
+    getTokensOfPublicLink: async function () {
```
Comment (dpakach) [inline tests/acceptance/pageObjects/FilesPageElement/publicLinksDialog.js]: The function name is not clear. It says `getTokens...` but infact it looks like it returns boolean value.

Comment (dpakach) [reply tests/acceptance/pageObjects/FilesPageElement/publicLinksDialog.js]: I think you should move this entire function out of the page object. This looks mostly like assertion so should not be in the page object.

Comment (jasson99) [reply tests/acceptance/pageObjects/FilesPageElement/publicLinksDialog.js]: done

**Comment by skshetry** [comment]
@kiranparajuli589, What was the intention of #2526? Does this PR resolve the issue?

**Comment by kiranparajuli589** [comment]
@skshetry, #2526 is just about adding acceptance test scenarios for mentioned issue.

File: tests/acceptance/features/webUISharingPublic/shareByPublicLink.feature [hunk 2]
```diff
@@ -644,3 +644,11 @@ Feature: Share by public link
...
     Then file "lorem.txt" should be listed on the webUI
     And file "lorem (2).txt" should be listed on the webUI
+
+  Scenario: user creates multiple public links with same name for the same file/folder
+    Given user "user1" has logged in using the webUI
+    When the user creates a new public link for folder "simple-folder" using the webUI with
+      | name | same_link_name |
+    And the user creates a new public link for folder "simple-folder" using the webUI with
+      | name | same_link_name |
+    Then the tokens should be different for the public links
```
Comment (skshetry) [inline tests/acceptance/features/webUISharingPublic/shareByPublicLink.feature]:
```suggestion
    Then all the tokens should be unique for the public links on the webUI
```

cc @phil-davis

File: tests/acceptance/stepDefinitions/publicLinkContext.js [hunk 3]
```diff
@@ -182,3 +182,16 @@ When('the user closes the public link details sidebar', function () {
...
+Then('the tokens should be different for the public links', async function () {
+  const tokens = []
+  let uniqueTokens = false
+  const publicLinkList = await client.page.FilesPageElement.publicLinksDialog().getPublicLinkList()
+  for (let element of publicLinkList) {
+    element = element.substring(0, element.lastIndexOf('|'))
+    element = element.substring(element.lastIndexOf('\n') + 1)
+    tokens.push(element)
+  }
+  uniqueTokens = tokens.length === (new Set(tokens).size)
```
Comment (skshetry) [inline tests/acceptance/stepDefinitions/publicLinkContext.js]:
```suggestion
  isUnique = tokens.length === (new Set(tokens).size)
```

```suggestion
  return assert.ok(tokens.length === (new Set(tokens).size))
```

File: tests/acceptance/stepDefinitions/publicLinkContext.js [hunk 4]
```diff
@@ -182,3 +182,16 @@ When('the user closes the public link details sidebar', function () {
     .filesList()
     .closeSidebar(100)
 })
+
+Then('the tokens should be different for the public links', async function () {
+  const tokens = []
+  let uniqueTokens = false
```
Comment (skshetry) [inline tests/acceptance/stepDefinitions/publicLinkContext.js]:
```suggestion
```

File: tests/acceptance/features/webUISharingPublic/shareByPublicLink.feature [hunk 5]
```diff
@@ -644,3 +644,11 @@ Feature: Share by public link
     When the user uploads file "lorem.txt" keeping both new and existing files using the webUI
     Then file "lorem.txt" should be listed on the webUI
     And file "lorem (2).txt" should be listed on the webUI
+
+  Scenario: user creates multiple public links with same name for the same file/folder
```
Comment (HariBhandari07) [inline tests/acceptance/features/webUISharingPublic/shareByPublicLink.feature]: Scenario says the public link for file/folder, but you only test for a folder. Can you please also add test for a file, if possible using Scenario Outline.

**Comment by HariBhandari07** [comment]
LGTM

File: tests/acceptance/stepDefinitions/publicLinkContext.js (hunk 3 above)
Comment (jasson99) [reply tests/acceptance/stepDefinitions/publicLinkContext.js]: done

File: tests/acceptance/stepDefinitions/publicLinkContext.js (hunk 4 above)
Comment (jasson99) [reply tests/acceptance/stepDefinitions/publicLinkContext.js]: done

**Comment by kiranparajuli589** [comment]
LGTM :+1:

**Pull request merged**